src/date-range-picker/range-picker.tsx

```diff
@@ -0,0 +1,159 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useEffect } from 'react';
+import Calendar, { DayIndex } from './calendar';
+import InternalBox from '../box/internal';
+import SpaceBetween from '../space-between/index.js';
+
+import RelativeRangePicker from './relative-range';
+import ModeSwitcher from './mode-switcher';
+import { useDateRangePicker } from './use-date-range-picker';
+import { DateRangePickerBaseProps } from './interfaces';
+import { fireNonCancelableEvent } from '../internal/events/index.js';
+import InternalAlert from '../alert/internal';
+import LiveRegion from '../internal/components/live-region';
+import styles from './styles.css.js';
+
+export interface DateRangePickerDropdownProps
+  extends Pick<
+    Required<DateRangePickerBaseProps>,
+    | 'value'
+    | 'relativeOptions'
+    | 'isDateEnabled'
+    | 'locale'
+    | 'isValidRange'
+    | 'i18nStrings'
+    | 'dateOnly'
+    | 'rangeSelectorMode'
+    | 'timeInputFormat'
+    | 'onChange'
+    | 'timeOffset'
+  > {
+  startOfWeek: DayIndex;
+  isSingleGrid: boolean;
+
+  ariaLabelledby?: string;
+  ariaDescribedby?: string;
+}
+
+export function DateRangePickerEmbedded({
+  value,
+  locale = '',
+  startOfWeek,
+  isDateEnabled = () => true,
+  relativeOptions,
+  isSingleGrid,
+  i18nStrings,
+  dateOnly,
+  timeInputFormat,
+  rangeSelectorMode,
+  onChange,
+  timeOffset,
+  isValidRange,
+}: DateRangePickerDropdownProps) {
+  const {
+    fillMissingTime,
+    rangeSelectionMode,
+    setRangeSelectionMode,
+    selectedAbsoluteRange,
+    setSelectedAbsoluteRange,
+    selectedRelativeRange,
+    setSelectedRelativeRange,
+    focusRefs,
+    validationResult,
+    setValidationResult,
+    VALID_RANGE,
+    formatValue,
+  } = useDateRangePicker({
+    value,
+    relativeOptions,
+    rangeSelectorMode,
+  });
+
+  useEffect(() => {
+    const newValue = rangeSelectionMode === 'relative' ? selectedRelativeRange : fillMissingTime(selectedAbsoluteRange);
+
+    if (newValue) {
+      const validationResult = isValidRange(newValue);
+      setValidationResult(validationResult || VALID_RANGE);
+      if (validationResult?.valid === false) {
+        setValidationResult(validationResult);
+      }
+    }
+
+    fireNonCancelableEvent(onChange, { value: formatValue(newValue, { dateOnly, timeOffset }) });
```

Comment:
Looks like it is this anti-pattern: https://beta.reactjs.org/learn/you-might-not-need-an-effect#notifying-parent-components-about-state-changes

Is there any chance to follow their recommendations? 

P.S. highly recommend reading the full article

Reply:
Thank you for that article, very applicable to this PR. Will incorporate the recommendations